src/pt/huntersscans/src/eu/kanade/tachiyomi/extension/pt/huntersscans/HuntersScans.kt

```diff
@@ -1,23 +1,247 @@
 package eu.kanade.tachiyomi.extension.pt.huntersscans
 
-import eu.kanade.tachiyomi.multisrc.madara.Madara
-import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import android.app.Application
+import android.content.SharedPreferences
+import android.util.Log
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.randomua.addRandomUAPreferenceToScreen
+import eu.kanade.tachiyomi.lib.randomua.getPrefCustomUA
+import eu.kanade.tachiyomi.lib.randomua.getPrefUAType
+import eu.kanade.tachiyomi.lib.randomua.setRandomUserAgent
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
 import okhttp3.OkHttpClient
-import java.text.SimpleDateFormat
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class HuntersScans : Madara(
-    "Hunters Scan",
-    "https://huntersscan.xyz/",
-    "pt-BR",
-    SimpleDateFormat("dd 'de' MMMMM 'de' yyyy", Locale("pt", "BR")),
-) {
+class HuntersScans : ParsedHttpSource(), ConfigurableSource {
+    override val name = "Hunters Scans"
 
-    override val client: OkHttpClient = super.client.newBuilder()
-        .rateLimit(1, 2, TimeUnit.SECONDS)
-        .build()
+    override val lang = "pt-BR"
 
-    override val useNewChapterEndpoint = true
-    override val useLoadMoreRequest = LoadMoreStrategy.Always
+    override val supportsLatest = true
+
+    override val baseUrl = "https://huntersscan.xyz"
+
+    override val id = 7310576253330700902
```

Comment:
why hard code the id? remove it.

also bump the `versionId` to force users to migrate to the newer urls

Reply:
I'll bump ```versionId```

Reply:
Did I make a mistake?

Reply:
If the URL format does not change after this theme migration, it's okay to keep the ID. You don't even need to bump `versionId` or override `id`.

In theory you could bump `versionId` but keep the `id` to indicate there had been a theme change but URL format hasn't changed.

### Explanation

Bumping `versionId` results in a different `id`, so it's useful when the URL format has changed (for example, old site uses `/manga/entry-name` but new site uses `/manhwa/entry_name`) and we need to force users to migrate their entries.

Keeping `id` only happens when you rename the source (`override val name = "XXX"` or `Madara("XXX", ...)` part) but nothing else changes. Since `id` is computed from `name`, `lang` and `versionId`, you need to keep the old `id`.

Reply:
Resolved: https://github.com/keiyoushi/extensions-source/pull/1650/commits/1a7026e76c332b00f214e0620563dddb0b22b3e7